merlin/systems/dag/ops/operator.py

```diff
@@ -306,8 +310,7 @@ def _schema_to_dict(schema: Schema) -> dict:
     return schema_dict
 
 
-def _add_model_param(params, paramclass, col_schema, dims=None):
-    dims = dims if dims is not None else [-1, 1]
+def add_model_param(params, paramclass, col_schema, dims=None):
```

Comment:
Computing the dims was a second responsibility of this function, so it moved to a separate utility function (above.)

Reply:
I like this fall back.